src/main/scala/zio/nio/ByteBuffer.scala

```diff
@@ -3,142 +3,142 @@ package zio.nio
 import zio.{ Chunk, IO, ZIO }
 import java.nio.{ BufferUnderflowException, ByteOrder, ReadOnlyBufferException, ByteBuffer => JByteBuffer }
 
-final class ByteBuffer(byteBuffer: JByteBuffer) extends Buffer[Byte](byteBuffer) {
+class ByteBuffer protected[nio] (byteBuffer: JByteBuffer) extends Buffer[Byte](byteBuffer) {
```

Comment:
I'm using ByteBuffer in zio-keeper https://github.com/zio/zio-keeper/pull/44 please keep it public or we could figure out better approach.

Reply:
Right, you're starting with a Java `ByteBuffer` you want to wrap? Yeah, I hadn't considered that case. Anyone have a preference between a public constructor for this, or putting:

```scala
def byte(javaBuffer: JByteBuffer): IO[Nothing, ByteBuffer] = new ByteBuffer(javaBuffer)
```

on the `Buffer` object?

Reply:
as I mentioned in other PR maybe we should add 
```
object ByteBuffer {
   def allocateDirect(size: Int) = new zio.nio.ByteBuffer(java.nio.ByteBuffer.allocateDirect(size))
   def fromChunk(chunk: Chunk[Byte]) = new zio.nio.ByteBuffer(java.nio.ByteBuffer.wrap(chunk.toArray))
}
```

what do you think?

Reply:
NVM in `Buffer` there are those methods I just missed them.

`https://github.com/zio/zio-nio/blob/master/src/main/scala/zio/nio/Buffer.scala#L107`